libretro/Makefile

```diff
@@ -56,7 +56,7 @@ ifneq (,$(findstring unix,$(platform)))
    CC = gcc
    SHARED := -shared -z defs
    else
-   SHARED := -shared -Wl,--version-script=link.T -Wl,-z,defs
+   SHARED := -shared -Wl,--version-script=link.T
```

Comment:
Without this, unresolved symbols will not be caught at compile time. Can we detect solaris in the makefile and remove it only in that case?

Reply:
I'll ask the @kwyxz

Reply:
We can definitely create a specific case for Solaris

Reply:
Wait isn't it already what we're doing here? 

`ifneq ($(findstring SunOS,$(shell uname -a)),)`

Am I looking at the wrong commit?

Reply:
My apologies if I missed it, I'll check in the morning

Reply:
@kwyxz thing is, the SHARED flags were changed for solaris but that is being used for all unixes apparently

Reply:
```Makefile
ifneq (,$(findstring unix,$(platform)))
   TARGET := $(TARGET_NAME)_libretro.so
   fpic := -fPIC
   ifneq ($(findstring SunOS,$(shell uname -a)),)
   CC = gcc
   SHARED := -shared -Wl,--version-script=link.T -Wl,-z,defs
   else
   SHARED := -shared -Wl,--version-script=link.T
   endif
   ifneq ($(findstring Haiku,$(shell uname -a)),)
      LIBS :=
   endif
```

I think this would be enough?

Reply:
or is it the other way around? not good on makefile fu

Reply:
I might be wrong but here's how I understand the current Makefile: the ifneq line is a double negative
````
   ifneq ($(findstring SunOS,$(shell uname -a)),)
   CC = gcc
   SHARED := -shared -z defs
   else
   SHARED := -shared -Wl,--version-script=link.T -Wl,-z,defs
   endif
````
If the findstring command looking for "SunOS" does not return "empty" it means we're building for Solaris hence `CC` becomes `gcc` and SHARED becomes `-shared -z defs` (gcc on Solaris does not handle commas for some reason)
Otherwise SHARED is `-shared -Wl,--version-script=link.T -Wl,-z,defs`

We could replace `uname -a` with `uname -s` though, to avoid hostnames that would include `SunOS` even though people doing this are looking for trouble IMHO